q-file-share-ui/src/app/dashboard/page.tsx

```diff
@@ -1,18 +1,155 @@
+"use client";
+
 import { NavBar } from "@/modules";
+import receivedLogo from "../assets/received-logo.png"
+import sharedLogo from "../assets/shared-logo.png"
+import shareLogo from "../assets/share-logo.png"
+import React from 'react';
+import { BrowserRouter as Router, Routes, Route, useNavigate } from 'react-router-dom';
+import type { StaticImageData } from 'next/image';
+
+interface CardProps {
+  title: string;
+  description: string;
+  logo: StaticImageData;
+  route: string;
+}
 
-const Dashboard = () => {
+const Dashboard: React.FC = () => {
   return (
-    <>
+    <Router>
       <NavBar
         showLogo={true}
         pages={[
           { name: "Shared Files", path: "/shared-files" },
           { name: "Received Files", path: "/received-files" },
         ]}
       />
-      <div>Secure File Share using Post-Quantum Cryptography</div>
-    </>
+      <div style={styles.dashboardContainer}>
+        {/* All cards in a single flex container */}
+        <div style={styles.cardContainer}>
+          <Card title="Share File" description="Securely share your files" logo={shareLogo} route="/share-file" />
+          <Card title="Received Files" description="View files you've received" logo={receivedLogo} route="/received-files" />
+          <Card title="Shared Files" description="Manage your shared files" logo={sharedLogo} route="/shared-files" />
+        </div>
+        <Activity />
+      </div>
+
+      {/* Define Routes */}
+      <Routes>
+        <Route path="/share-file" element={<ShareFile />} />
+        <Route path="/received-files" element={<ReceivedFiles />} />
```

Comment:
Remove the react router dom packages. Use the redirect from next/navigation package.